controller/search_blackbox_test.go

```diff
@@ -343,3 +353,88 @@ func (s *searchBlackBoxTest) TestAutoRegisterHostURL() {
 	queryString2 := fmt.Sprintf("http://%s/work-item/board/detail/%s", customHost2, *wi.Data.ID)
 	s.verifySearchByKnownURLs(wi, customHost2, queryString2)
 }
+
+func (s *searchBlackBoxTest) TestSearchWorkItemsSpaceContext() {
+	name1 := "Ultimate Space 1" + uuid.NewV4().String()
+	var space1 *space.Space
+	application.Transactional(s.db, func(app application.Application) error {
+		sp := space.Space{
+			Name: name1,
+		}
+		var err error
+		space1, err = app.Spaces().Create(context.Background(), &sp)
+		require.Nil(s.T(), err)
+		return nil
+	})
+
+	name2 := "Ultimate Space 2" + uuid.NewV4().String()
+	var space2 *space.Space
+	application.Transactional(s.db, func(app application.Application) error {
+		sp := space.Space{
+			Name: name2,
+		}
+		var err error
+		space2, err = app.Spaces().Create(context.Background(), &sp)
+		require.Nil(s.T(), err)
+		return nil
+	})
+
+	// WI for space 1
+	for i := 0; i < 3; i++ {
+		wi, err := s.wiRepo.Create(
+			s.ctx,
+			space1.ID,
+			workitem.SystemBug,
+			map[string]interface{}{
+				workitem.SystemTitle:       "shutter_island common_word random - " + uuid.NewV4().String(),
+				workitem.SystemDescription: nil,
+				workitem.SystemCreator:     "pranav",
+				workitem.SystemState:       workitem.SystemStateClosed,
+			},
+			s.testIdentity.ID)
+		require.Nil(s.T(), err)
+		require.NotNil(s.T(), wi)
+	}
+	// WI for space 2
+	for i := 0; i < 5; i++ {
+		wi, err := s.wiRepo.Create(
+			s.ctx,
+			space2.ID,
+			workitem.SystemBug,
+			map[string]interface{}{
+				workitem.SystemTitle:       "inception common_word random - " + uuid.NewV4().String(),
+				workitem.SystemDescription: nil,
+				workitem.SystemCreator:     "pranav",
+				workitem.SystemState:       workitem.SystemStateClosed,
+			},
+			s.testIdentity.ID)
+		require.Nil(s.T(), err)
+		require.NotNil(s.T(), wi)
+	}
+	// when
+	q := "common_word"
+	space1IDStr := space1.ID.String()
+	_, sr := test.ShowSearchOK(s.T(), nil, nil, s.controller, nil, nil, q, &space1IDStr)
+	// then
+	require.NotEmpty(s.T(), sr.Data)
+	assert.Len(s.T(), sr.Data, 3)
+	for _, item := range sr.Data {
+		// make sure that retrived items are from space 1 only
+		assert.Contains(s.T(), item.Attributes[workitem.SystemTitle], "shutter_island common_word")
+	}
+	space2IDStr := space2.ID.String()
+	_, sr = test.ShowSearchOK(s.T(), nil, nil, s.controller, nil, nil, q, &space2IDStr)
+	// then
+	require.NotEmpty(s.T(), sr.Data)
+	assert.Len(s.T(), sr.Data, 5)
+	for _, item := range sr.Data {
+		// make sure that retrived items are from space 2 only
+		assert.Contains(s.T(), item.Attributes[workitem.SystemTitle], "inception common_word")
+	}
+
+	// when searched without spaceID then it should get all related WI
```

Comment:
Could you separate this out to a different test ?
Since this is critical in ensuring that this is not a breaking change.